contracts/src/rollup/BOLDUpgradeAction.sol

```diff
@@ -348,14 +348,17 @@ contract BOLDUpgradeAction {
         if (stakerCount > 50) {
             stakerCount = 50;
         }
-        for (uint64 i = 0; i < stakerCount; i++) {
+        for (uint64 i = 0; i < stakerCount;) {
             address stakerAddr = ROLLUP_READER.getStakerAddress(i);
             OldStaker memory staker = ROLLUP_READER.getStaker(stakerAddr);
             if (staker.isStaked && staker.currentChallenge == 0) {
                 address[] memory stakersToRefund = new address[](1);
                 stakersToRefund[0] = stakerAddr;
 
                 IOldRollupAdmin(address(OLD_ROLLUP)).forceRefundStaker(stakersToRefund);
+                stakerCount -= 1;
```

Comment:
We should add a comment to state that `forceRefundStaker` removes a staker from the internal list - so we should update the stakerCount param to reflect.

We should probably also add a comment in the i++ section explaining that we leave this staker in the contract and move past it.